### PR TITLE
ci: limit interop tests runs to core folders

### PR DESCRIPTION
**Pull request by p-shahi**
Avoid running the interop tests on code changes to directories that do not need testing.
Ref: https://github.com/libp2p/test-plans/pull/272

### CODE REVIEW

File: .github/workflows/interop-test.yml [hunk 1]
```diff
@@ -1,9 +1,22 @@
 name: Interoperability Testing
 on:
+  workflow_dispatch:
```
Comment (marten-seemann) [inline .github/workflows/interop-test.yml:3]: What is this doing?

**Review by marten-seemann — COMMENTED** [review]
Would it make sense to use a blacklist instead of a whitelist here? I assume you want to exclude runs for changes to the README and similar files.

File: .github/workflows/interop-test.yml (hunk 1 above)
Comment (p-shahi) [reply .github/workflows/interop-test.yml:3]: it enables a workflow to be triggered [manually](https://docs.github.com/en/actions/using-workflows/manually-running-a-workflow#configuring-a-workflow-to-run-manually).

**Comment by p-shahi** [comment]
> Would it make sense to use a blacklist instead of a whitelist here? I assume you want to exclude runs for changes to the README and similar files.

The way its configured now with the allowlist, it won't run for top level directory changes either. if your preference is to have a deny list I can make that change.

**Review by MarcoPolo — APPROVED** [review]
No preference for allow vs deny.

**Pull request merged**